<launch>

  <arg name="rosbag" default="true" />
  <arg name="integrate_color" default="true" />

  <group if="$(arg rosbag)">
    <include file="$(find jsk_pcl_ros)/sample/include/play_rosbag_room73b2_table.xml" />
  </group>

  <arg name="INPUT_CAMERA_INFO" default="/camera/depth_registered/camera_info" />
  <arg name="INPUT_DEPTH" default="/camera/depth_registered/hw_registered/image_rect_raw" />
  <arg name="INPUT_COLOR" default="/camera/rgb/image_rect_color" />

  <node name="rosconsole_overlay_text"
        pkg="jsk_rviz_plugins" type="rosconsole_overlay_text.py">
    <rosparam>
      nodes:
        - /kinfu
    </rosparam>
  </node>
  <node name="rviz"
        pkg="rviz" type="rviz"
        args="-d $(find jsk_pcl_ros)/sample/rviz/kinfu.rviz">
  </node>

  <node name="kinfu"
        pkg="nodelet" type="nodelet"
        args="standalone jsk_pcl/Kinfu">
    <remap from="~input/camera_info" to="$(arg INPUT_CAMERA_INFO)" />
    <remap from="~input/depth" to="$(arg INPUT_DEPTH)" />
    <remap from="~input/color" to="$(arg INPUT_COLOR)" />
    <rosparam subst_value="true">
      integrate_color: $(arg integrate_color)
    </rosparam>
  </node>

</launch>
